pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/KinesisConsumer.java

```diff
@@ -155,6 +162,17 @@ private KinesisRecordsBatch getResult(StreamPartitionMsgOffset startOffset, Stre
         if (Thread.interrupted()) {
           break;
         }
+
+        if (requestSentTime == currentWindow && getRecordsResponse.records().isEmpty()) {
```

Comment:
@navina I feel like I should remove this isEmpty check from here. Reason being it doesn't cover the cases when we might be getting only 1-2 records on each .getRecords request and thus exceed the throughput limit.

Reply:
ok yeah makes sense 👍